activitysim/abm/models/vehicle_type_choice.py

```diff
@@ -20,14 +20,313 @@
+        pipeline.replace_table("households", households)
+
+    # - annotate persons table
+    if model_settings.get('annotate_persons'):
+        persons = inject.get_table('persons').to_frame()
+        expressions.assign_columns(
+            df=persons,
+            model_settings=model_settings.get('annotate_persons'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_persons'))
+        pipeline.replace_table("persons", persons)
+
+    # - annotate vehicles table
+    if model_settings.get('annotate_vehicles'):
+        vehicles = inject.get_table('vehicles').to_frame()
+        expressions.assign_columns(
+            df=vehicles,
+            model_settings=model_settings.get('annotate_vehicles'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_vehicles'))
+        pipeline.replace_table("vehicles", vehicles)
+
+
+def iterate_vehicle_type_choice(
+        vehicles_merged,
+        model_settings,
+        model_spec,
+        locals_dict,
+        estimator,
+        chunk_size,
+        trace_label):
+    """
+    Iterates through household vehicle numbers and selects a vehicle type of
+    the form {body_type}_{age}_{fuel_type}. The preprocessor is run for each
+    iteration on the entire chooser table, not just the one for the current
+    vehicle number.  This allows for computation of terms involving the presence
+    of other household vehicles.
+
+    Vehicle type data is read in according to the specification and joined to
+    the alternatives. It can optionally be included in the output vehicles table
+    by specifying the `COLS_TO_INCLUDE_IN_VEHICLE_TABLE` option in the model yaml.
+
+    Parameters
+    ----------
+    vehicles_merged : orca.DataFrameWrapper
+    model_settings : dict
+        yaml model settings file as dict
+    model_spec : pandas.DataFrame
+        omnibus spec file with expressions in index and one column per segment
+    locals_dict : dict
+    estimator : Estimator object
+    chunk_size : orca.injectable
+    trace_label : str
+
+    Returns
+    -------
+    all_choices : pandas.DataFrame
+        table of selected vehicle types and associated data
+    all_choosers : pandas.DataFrame
+        table of chooser data with preprocessor variables included
+    """
+    # - model settings
+    nest_spec = config.get_logit_model_settings(model_settings)
+    vehicle_type_data_file = model_settings.get('VEHICLE_TYPE_DATA_FILE', None)
+    probs_spec_file = model_settings.get("PROBS_SPEC", None)
+    alts_cats_dict = model_settings.get('combinatorial_alts', False)
+
+    # adding vehicle type data to be available to locals_dict regardless of option
+    if vehicle_type_data_file:
+        vehicle_type_data = pd.read_csv(
+            config.config_file_path(vehicle_type_data_file), comment='#')
+        fleet_year = model_settings.get('FLEET_YEAR')
+
+        vehicle_type_data['age'] = (1 + fleet_year - vehicle_type_data['vehicle_year']).astype(str)
```

Comment:
It seems to make sense that we would be able to imply a variation in fuel type based on vehicle age. However, in this case age isn't necessarily relative.  A 10 year old car in a 2030 scenario should be equal to a 2 year old car in 2022. 

I'm trying to figure out how this is accommodated within the probabilities but can't seem to find anywhere that this is done?

Reply:
Yeah, this wasn't done very consistently... I changed the probabilities file to include this same calculation -- the probabilities file now includes body_type and vehicle_year columns just like the vehicle_type_data.csv file and age is explicitly calculated based on the user input fleet_year.